learnify/mobile/lib/features/auth/forget_password/view-model/forget_password_view_model.dart

```diff
@@ -0,0 +1,55 @@
+import 'package:async/async.dart';
+import 'package:flutter/material.dart';
+
+import '../../../../core/base/view-model/base_view_model.dart';
+
+class ForgetPasswordViewModel extends BaseViewModel {
+  late TextEditingController _emailController;
+
+  TextEditingController get emailController => _emailController;
+
+  late GlobalKey<FormState> _formKey;
+  GlobalKey<FormState> get formKey => _formKey;
+
+  bool _canVerify = false;
+  bool get canVerify => _canVerify;
+
+  @override
+  void initViewModel() {}
+
+  @override
+  void initView() {
+    _formKey = GlobalKey<FormState>();
+    _emailController = TextEditingController();
+    _emailController.addListener(_controllerListener);
+  }
+
+  @override
+  void disposeView() {
+    _emailController.dispose();
+    super.disposeView();
+  }
+
+  void _controllerListener() {
+    final bool newCanVerify = _emailController.text.isNotEmpty;
+    if (_canVerify == newCanVerify) return;
+    _canVerify = newCanVerify;
+    notifyListeners();
+  }
+
+  Future<String?> forgetPassword() async {
+    await operation?.cancel();
+    operation =
+        CancelableOperation<String?>.fromFuture(_forgetPasswordRequest());
+    final String? res = await operation?.valueOrCancellation();
+    return res;
+  }
+
+  Future<String?> _forgetPasswordRequest() async {
+    final bool isValid = formKey.currentState?.validate() ?? false;
+    if (isValid) {
+      return null;
```

Comment:
just remove this line and add a `return null;` line at the end of this method. We changed this logic, you can see the updated version in the "dev" branch. Error is already displayed under the email field. So, we don't need to show a dialog for form validation errors.